data-runtime/src/main/java/io/micronaut/data/runtime/intercept/criteria/AbstractSpecificationInterceptor.java

```diff
@@ -205,11 +207,15 @@ private <E> StoredQuery<E, Object> buildFind(MethodInvocationContext<T, R> conte
                 }
             }
         }
-        QueryResult queryResult = ((QueryResultPersistentEntityCriteriaQuery) criteriaQuery).buildQuery(sqlQueryBuilder);
+        QueryResultPersistentEntityCriteriaQuery queryModelCriteriaQuery = (QueryResultPersistentEntityCriteriaQuery) criteriaQuery;
+        Collection<JoinPath> joinPaths = queryModelCriteriaQuery.getQueryModel().getJoinPaths();
+        QueryResult queryResult = queryModelCriteriaQuery.buildQuery(sqlQueryBuilder);
```

Comment:
This one can be replaced with:
```java
queryBuilder.buildQuery(queryModel)
```
So you don't create the model twice.

Reply:
Done